### PR TITLE
New: OnGrab Notifications

### LINKED ISSUE #958: allow running custom script on grab
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe

I want AddqBittorrentTrackers.sh from https://github.com/Jorman/Scripts to run after grabbing but Prowlarr doesn't allow it

### Describe the solution you'd like

option to run custom script after grabbing like sonarr and radarr

### Describe alternatives you've considered

can't think of any

### Anything else?

no

[AB#2996](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/2996)

### PR DESCRIPTION
**Pull request by Qstick**
#### Database Migration
YES - 029 (Grab Columns on Notifications Table)

#### Description
Notifications for grabs in Prowlarr (option to ignore manual grabs). Also adds UI header `X-Prowlarr-Client` to identify if requests are made from UI. 

#### Todos
- [x] Tests
- [x] Translation Keys (./src/NzbDrone.Core/Localization/Core/en.json)
- [x] [Wiki Updates](https://wiki.servarr.com)

Fixes #958

### CODE REVIEW

File: src/NzbDrone.Core/Localization/Core/en.json [hunk 1]
```diff
@@ -185,6 +185,7 @@
   "IgnoredAddresses": "Ignored Addresses",
   "IllRestartLater": "I'll restart later",
   "IncludeHealthWarningsHelpText": "Include Health Warnings",
+  "IncludeManualGrabsHelpText": "Include Manual Grabs",
```
Comment (bakerboy448) [inline src/NzbDrone.Core/Localization/Core/en.json:188]:
```suggestion
  "IncludeManualGrabsHelpText": "Include Manual Grabs made within Prowlarr",
```

File: src/NzbDrone.Core/Localization/Core/en.json [hunk 2]
```diff
@@ -272,7 +273,8 @@
   "Ok": "Ok",
   "OnApplicationUpdate": "On Application Update",
   "OnApplicationUpdateHelpText": "On Application Update",
-  "OnGrab": "On Grab",
+  "OnGrab": "On Release Grab",
+  "OnGrabHelpText": "On Release Grab",
```
Comment (bakerboy448) [inline src/NzbDrone.Core/Localization/Core/en.json:277]:
```suggestion
  "OnGrabHelpText": "On Release Grab by another application via Prowlarr's API",
```

File: src/NzbDrone.Core/Notifications/NotificationService.cs [hunk 3]
```diff
@@ -35,6 +41,43 @@ private bool ShouldHandleHealthFailure(HealthCheck.HealthCheck healthCheck, bool
...
         }
 
+        private bool ShouldHandleOnGrab(GrabMessage message, bool includeManual)
+        {
+            if (message.GrabTrigger == GrabTrigger.Api)
+            {
+                return true;
+            }
+
+            if (message.GrabTrigger == GrabTrigger.Api && includeManual)
```
Comment (bakerboy448) [inline src/NzbDrone.Core/Notifications/NotificationService.cs:51]:
```suggestion
            if (message.GrabTrigger == GrabTrigger.Manual && includeManual)
```

File: src/NzbDrone.Core/Indexers/Events/IndexerDownloadEvent.cs [hunk 4]
```diff
@@ -1,26 +1,37 @@
...
-        public IndexerDownloadEvent(int indexerId, bool successful, string source, string host, string title, string url, bool redirect = false)
+        public IndexerDownloadEvent(ReleaseInfo release, bool successful, string source, string host, string title, string url)
         {
-            IndexerId = indexerId;
+            Release = release;
             Successful = successful;
             Source = source;
             Host = host;
             Title = title;
-            Redirect = redirect;
```
Comment (bakerboy448) [inline src/NzbDrone.Core/Indexers/Events/IndexerDownloadEvent.cs:22]: is dropping redirect here intentional?

Comment (Qstick) [reply src/NzbDrone.Core/Indexers/Events/IndexerDownloadEvent.cs:22]: Yes, it gets set manually for the one instance it's used

File: src/NzbDrone.Core/Localization/Core/en.json (hunk 2 above)
Comment (Qstick) [reply src/NzbDrone.Core/Localization/Core/en.json:277]: Keeping this one as it aligns with the others

File: src/Prowlarr.Api.V1/Search/ReleaseResource.cs [hunk 5]
```diff
@@ -36,6 +37,21 @@ public class ReleaseResource : RestResource
         public int? Seeders { get; set; }
         public int? Leechers { get; set; }
         public DownloadProtocol Protocol { get; set; }
+
+        public string FileName
+        {
+            get
+            {
+                var extension = "torrent";
```
Comment (bakerboy448) [inline src/Prowlarr.Api.V1/Search/ReleaseResource.cs:45]: Magnets?

Comment (Qstick) [reply src/Prowlarr.Api.V1/Search/ReleaseResource.cs:45]: They should fail similar to now when save is clicked in the UI, shouldn't be any change in behavior on that front.

**Pull request merged**